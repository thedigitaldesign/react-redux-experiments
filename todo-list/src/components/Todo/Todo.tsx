import React, { Component } from 'react'

export class Todo extends Component {
  render() {
    return (
      <>
        <div className="field">
          <label className="label">Title</label>
          <div className="control">
            <input type="text" className="input input-title" placeholder="Todo task" />
          </div>
        </div>

        <div className="field">
          <label className="label">Description</label>
          <div className="control">
            <textarea className="textarea textarea-description" placeholder="Todo description"></textarea>
          </div>
        </div>
      </>
    )
  }
}
