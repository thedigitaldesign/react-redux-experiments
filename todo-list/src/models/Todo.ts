export interface Todo {
  id: string | number
  title: string
  description?: string
}