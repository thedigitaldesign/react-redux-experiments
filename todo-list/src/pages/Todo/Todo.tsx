import React, { Component } from 'react'

// Components
import { TodoList } from '../../components'

export default class Todo extends Component {
  render() {
    return (
      <>
        <TodoList />
      </>
    )
  }
}
