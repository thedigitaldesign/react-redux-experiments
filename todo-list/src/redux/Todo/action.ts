// Models
import { Todo } from '../../models/Todo'

// Types
import * as type from '../types'

export const AddTodoItem = (item: Todo) => {
  return {
    type: type.TodoItemTypes.TODO_ITEM,
    payload: item,
  }
}
