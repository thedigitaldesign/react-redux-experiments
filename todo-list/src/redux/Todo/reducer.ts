// Models
import { Todo } from '../../models'

export default (state: string, action: { type: string, payload: Todo }) => {
  return {}
}
