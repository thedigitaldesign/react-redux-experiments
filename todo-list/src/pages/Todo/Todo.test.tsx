import React from 'react'

// Packages
import Enzyme, {shallow} from 'enzyme'

// Routes
import Todo from './Todo'

describe('Pages > Todo', () => {
  const todo = shallow(<Todo />)

  it('should render pages Todo', () => {
    expect(todo).toMatchSnapshot()
  });
});
