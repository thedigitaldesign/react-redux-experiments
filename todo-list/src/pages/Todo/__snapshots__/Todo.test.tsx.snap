// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages > Todo should render pages Todo 1`] = `ShallowWrapper {}`;
